src/components/ImageView/index.native.js

```diff
@@ -102,148 +115,123 @@ class ImageView extends PureComponent {
+    const shouldShowLoadingIndicator = isLoading || !hasImageDimensions;
+
+    // Zoom view should be loaded only after measuring actual image dimensions, otherwise it causes blurred images on Android
+    return (
+        <View
+            style={[styles.w100, styles.h100, styles.alignItemsCenter, styles.justifyContentCenter, styles.overflowHidden]}
+            onLayout={(event) => {
+                const layout = event.nativeEvent.layout;
+                setContainerHeight(layout.height);
+            }}
+        >
+            {Boolean(containerHeight) && (
+                <ImageZoom
+                    ref={zoom}
+                    onClick={onPress}
+                    cropWidth={windowWidth}
+                    cropHeight={calculatedWindowHeight}
+                    imageWidth={imageWidth}
+                    imageHeight={imageHeight}
+                    onStartShouldSetPanResponder={onStartShouldSetPanResponder}
+                    onMove={({scale}) => {
+                        onScaleChanged(scale);
+                        imageZoomScale.current = scale;
+                    }}
+                >
+                    <Image
+                        style={[
+                            styles.w100,
+                            styles.h100,
+                            style,
+
+                            // Hide image while loading so ImageZoom can get the image
+                            // size before presenting - preventing visual glitches or shift
+                            // due to ImageZoom
+                            shouldShowLoadingIndicator ? styles.opacity0 : styles.opacity1,
+                        ]}
+                        source={{uri: url}}
+                        isAuthTokenRequired={isAuthTokenRequired}
+                        resizeMode={Image.resizeMode.contain}
+                        onLoadStart={imageLoadingStart}
+                        onLoad={configureImageZoom}
+                    />
+                    {/**
+                       Create an invisible view on top of the image so we can capture and set the amount of touches before
+                      the ImageZoom's PanResponder does. Children will be triggered first, so this needs to be inside the
+                      ImageZoom to work
+                      */}
+                    <View
+                        /* eslint-disable-next-line react/jsx-props-no-spreading */
+                        {...panResponder.panHandlers}
+                        style={[styles.w100, styles.h100, styles.invisible]}
+                    />
+                </ImageZoom>
+            )}
+            {shouldShowLoadingIndicator && <FullscreenLoadingIndicator style={[styles.opacity1, styles.bgTransparent]} />}
+        </View>
+    );
 }
 
 ImageView.propTypes = propTypes;
 ImageView.defaultProps = defaultProps;
+ImageView.displayName = 'ImageView';
 
-export default withWindowDimensions(ImageView);
+export default React.memo(ImageView);
```

Comment:
is it needed? 

because the `style` will be a new object always, so the memo won't work
also the `onScaleChanged` and `onPress` are not stable will change always, I think in this case won't benefit the component